accounts-db/src/accounts_db.rs

```diff
@@ -4006,6 +4020,18 @@ impl AccountsDb {
                         unrefed_pubkeys.push(pubkey);
                         result = AccountsIndexScanResult::Unref;
                         dead += 1;
+
+                        // If we are marking something dead, and the only remaining alive account is zero lamport, then make that zero lamport slot ready to be cleaned.
+                        // If that slot happens to only contain zero lamport accounts, the whole slot will go away
+                        if slot_list.len() == 1 // should we also check for ref counts here?
```

Comment:
I think we should check for ref count. However, we should check after we have unref'd this one. There can be 3 refs, 1 alive that is zero lamport. Each time we unref a pubkey in shrink where the alive one is dead, we should check if refcount has become 1. If so, we know this account can be marked dead and shrunk or entire slot cleaned.

Reply:
also, see:
https://github.com/anza-xyz/agave/pull/2865/commits/4bb2d9120abe1c7f0bb779fa0aa754976bb36d53